invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/search/components.js

```diff
@@ -25,80 +25,15 @@ import {
   Header,
   Icon,
   Input,
-  Item,
   Label,
   Message,
   Segment,
 } from "semantic-ui-react";
-import { SearchItemCreators } from "../utils";
+import RecordsResultsListItem from "../components/RecordsResultsListItem";
 import PropTypes from "prop-types";
 
 export const RDMRecordResultsListItem = ({ result }) => {
-  const accessStatusId = _get(result, "ui.access_status.id", "open");
-  const accessStatus = _get(result, "ui.access_status.title_l10n", "Open");
-  const accessStatusIcon = _get(result, "ui.access_status.icon", "unlock");
-  const createdDate = _get(
-    result,
-    "ui.created_date_l10n_long",
-    "No creation date found."
-  );
-  const creators = result.ui.creators.creators.slice(0, 3);
-
-  const descriptionStripped = _get(result, "ui.description_stripped", "No description");
-
-  const publicationDate = _get(
-    result,
-    "ui.publication_date_l10n_long",
-    "No publication date found."
-  );
-  const resourceType = _get(result, "ui.resource_type.title_l10n", "No resource type");
-  const subjects = _get(result, "ui.subjects", []);
-  const title = _get(result, "metadata.title", "No title");
-  const version = _get(result, "ui.version", null);
-
-  // Derivatives
-  const viewLink = `/records/${result.id}`;
-  return (
-    <Item>
-      <Item.Content>
-        <Item.Extra className="labels-actions">
-          <Label size="tiny" className="primary">
-            {publicationDate} ({version})
-          </Label>
-          <Label size="tiny" className="neutral">
-            {resourceType}
-          </Label>
-          <Label size="tiny" className={`access-status ${accessStatusId}`}>
-            {accessStatusIcon && <i className={`icon ${accessStatusIcon}`} />}
-            {accessStatus}
-          </Label>
-        </Item.Extra>
-        <Item.Header as="h2">
-          <a href={viewLink}>{title}</a>
-        </Item.Header>
-        <Item className="creatibutors">
-          <SearchItemCreators creators={creators} />
-        </Item>
-        <Item.Description>
-          {_truncate(descriptionStripped, { length: 350 })}
-        </Item.Description>
-        <Item.Extra>
-          {subjects.map((subject) => (
-            <Label key={subject.title_l10n} size="tiny">
-              {subject.title_l10n}
-            </Label>
-          ))}
-          {createdDate && (
-            <div>
-              <small>
-                {i18next.t("Uploaded on")} <span>{createdDate}</span>
-              </small>
-            </div>
-          )}
-        </Item.Extra>
-      </Item.Content>
-    </Item>
-  );
+  return <RecordsResultsListItem result={result} />;
```

Comment:
why do we need this wrapping component?

Reply:
My rationale behind this change was the following:

Initially we were using a functional component called RDMRecordsResultsListItem for both the results list in the frontpage and our custom RDM list of records in the search. For the frontpage, we were just using the RDMRecordsResultsListComponent for each record, while for the search we were using that component to override the ResultsList.item component.

I wanted to make some differentiation between the two use cases, as they may vary so, as part of the refactor I was doing to make RDMRecordsResultsListItem a React component (instead of a functional component), I divided it into two components. One component  is RecordsResultsListItem (https://github.com/inveniosoftware/invenio-app-rdm/pull/1971/files#diff-3c47a1ea3137a52366750ec052a4d5b1e6c685a8c5e441f851b95c7b00f24bf0) which is used by the recent uploads section, the other is RDMRecordsResultsListItem(invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/frontpage/RecordsList.js) which is used for the override of the uploads search page. This second component wraps the first one to avoid code duplciation because, in our RDM instance, they are equal.